collector/systemd_linux.go

```diff
@@ -424,3 +453,23 @@ func filterUnits(units []unit, whitelistPattern, blacklistPattern *regexp.Regexp
 
 	return filtered
 }
+
+func getSystemdVersion() int {
+	conn, err := newSystemdDbusConn()
+	if err != nil {
+		log.Warnf("Unable to get systemd dbus connection, defaulting systemd version to 0: %s", err)
+		return 0
+	}
+	defer conn.Close()
+	version, err := conn.GetManagerProperty("Version")
+	if err != nil {
+		log.Warn("Unable to get systemd version property, defaulting to 0")
+		return 0
+	}
+	version = strings.Replace(version, "\"", "", 2)
+	v, err := strconv.Atoi(version)
+	if err != nil {
+		log.Warnf("Got invalid systemd version: %v", version)
+	}
+	return v
```

Comment:
I think that the value of `v` here may not be sane when `err != nil`.

Reply:
`strconv.Atoi` should always return a `0` when `err != nil` so it should be ok, but it might make the code more clear to just return a 0 directly.  I'll update it.